### PR TITLE
New packages: php8.[01]*, xdebug8.[01], composer8.[01]

### PR DESCRIPTION
**Pull request by TinfoilSubmarine**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

Not sure what else is needed for this to round a complete PHP environment, but I am currently using the php8.0-* stuff to self-host Nextcloud and Firefly III without any issues.

~~In regards to 8.0.x vs 8.1.x, I opted for 8.0.x because in my use case both Nextcloud and Firefly III still haven't properly supported PHP 8.1.x. For reference, it's been out for [3 months](https://www.php.net/supported-versions.php). Perhaps we can have a php8.1 package as well? idk~~

EDIT: this would also close https://github.com/void-linux/void-packages/issues/31678

### CODE REVIEW

**Comment by TinfoilSubmarine** [comment]
Ah, forgot to look into that, I manually downloaded the distfiles on my machine.

This is some sort of issue with `xbps-fetch`...

**Comment by TinfoilSubmarine** [comment]
I added the 2 php-* packages left (ast and igbinary). 

The only other thing I use is composer, which I'm having a hard time making it play nice with the php binary being named something other than `php` in path.

I know that the [previous attempt](https://github.com/void-linux/void-packages/pull/31668) to get PHP 8 into repo had xdebug as well, so I'll look into that.

**Comment by TinfoilSubmarine** [comment]
I think everything is here from the list in https://github.com/void-linux/void-packages/issues/31678 in other than `pear`.

File: srcpkgs/php8.0-apcu/template
Comment (paper42) [inline srcpkgs/php8.0-apcu/template:11]: Why is this version restriction necessary?

**Review by paper42 — COMMENTED** [review]
some packages could be updated now, some need an update file to fix update check
```
NO VERSION found for xdebug8.1
composer8.1-2.2.7 -> composer8.1-2.2.8
composer8.1-2.2.7 -> composer8.1-2.2.9
NO VERSION found for php8.1-imagick
NO VERSION found for php8.1-apcu
NO VERSION found for php8.1
NO VERSION found for xdebug8.0
composer8.0-2.2.7 -> composer8.0-2.2.8
composer8.0-2.2.7 -> composer8.0-2.2.9
NO VERSION found for php8.0-imagick
NO VERSION found for php8.0
```

File: srcpkgs/php8.0-apcu/template
Comment (TinfoilSubmarine) [reply srcpkgs/php8.0-apcu/template]: Dropped version restrictions.

**Comment by TinfoilSubmarine** [comment]
> some packages could be updated now, some need an update file to fix update check
> 

All should be fixed and updated to newest versions now.

**Comment by paper42** [comment]
It would be nice if the packages had a maintainer set before merging them, would you like write your name there?

**Comment by TinfoilSubmarine** [comment]
> It would be nice if the packages had a maintainer set before merging them, would you like write your name there?

Done

**Review by paper42 — APPROVED** [review]

**Comment by jchook** [comment]
What's the preferred way to "choose" a version of php to be symlinked to `/usr/bin/php`?

Is there a way to do this with xbps-alternatives?

**Comment by jchook** [comment]
What's the preferred way to "choose" a version of php to be symlinked to `/usr/bin/php`?

Is there a way to do this with xbps-alternatives?

**Comment by TinfoilSubmarine** [comment]
I did not set that up in this PR; I can look into it if there is interest. Aa you mentioned, would just need to add xbps-alternatives to the package template(s).

> What's the preferred way to "choose" a version of php to be symlinked to `/usr/bin/php`?
> 
> Is there a way to do this with xbps-alternatives?

**Comment by paper42** [comment]
> What's the preferred way to "choose" a version of php to be symlinked to `/usr/bin/php`?
> 
> Is there a way to do this with xbps-alternatives?

Currently there isn't a way to do this, if you want to use php8.0, you have to type php8.0.

**Pull request merged**
